# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../lirc/linux-2.6.18.patch
# Copyright (C) 2006 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

--- lirc-0.8.0/drivers/lirc_dev/lirc_dev.c.orig	2006-10-06 16:45:05.000000000 -0400
+++ lirc-0.8.0/drivers/lirc_dev/lirc_dev.c	2006-10-06 16:54:18.000000000 -0400
@@ -51,7 +51,11 @@
 #include <linux/unistd.h>
 /* DevFS header */
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,4,0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,18)
 #include <linux/devfs_fs_kernel.h>
+#else
+#undef LIRC_HAVE_DEVFS_24
+#endif
 #endif
 /* SysFS header */
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)
--- ./drivers/lirc_imon/lirc_imon.c.orig	2006-10-06 16:55:24.000000000 -0400
+++ ./drivers/lirc_imon/lirc_imon.c	2006-10-06 16:55:53.000000000 -0400
@@ -58,7 +58,6 @@
 #include <linux/slab.h>
 #include <asm/uaccess.h>
 #include <linux/usb.h>
-#include <linux/devfs_fs_kernel.h>
 
 #include "drivers/lirc.h"
 #include "drivers/kcompat.h"
--- ./drivers/lirc_sasem/lirc_sasem.c.orig	2006-10-06 17:05:53.000000000 -0400
+++ ./drivers/lirc_sasem/lirc_sasem.c	2006-10-06 17:06:14.000000000 -0400
@@ -67,7 +67,6 @@
 #include <linux/slab.h>
 #include <asm/uaccess.h>
 #include <linux/usb.h>
-#include <linux/devfs_fs_kernel.h>
 
 #include "drivers/lirc.h"
 #include "drivers/kcompat.h"
